src/test/java/sirius/kernel/commons/CSVWriterSpec.groovy

```diff
@@ -57,4 +70,78 @@ class CSVWriterSpec extends BaseSpecification {
         output.toString() == '"a;b\\"";\\\\;c'
     }
 
+    def "escaping of separator with escape-char works if there is no quotation-char"() {
+        given:
+        StringWriter output = new StringWriter()
+        and:
+        CSVWriter writer = new CSVWriter(output)
+        writer.withQuotation('\0' as char)
+        when:
+        writer.writeArray("a;b")
+        and:
+        output.close()
+        then:
+        output.toString() == 'a\\;b'
+    }
+
+    def "throw an exception if we have to escape quotes, but there is no escape-char"() {
+        given:
+        StringWriter output = new StringWriter()
+        and:
+        CSVWriter writer = new CSVWriter(output)
+        writer.withEscape('\0' as char)
+        when:
+        def thrownException
+        try {
+            writer.writeArray('"a";b')
+        } catch (Exception e) {
+            thrownException = e
+        }
+        and:
+        output.close()
+        then:
+        thrownException instanceof IllegalArgumentException
+        thrownException.getMessage() == "Cannot output a quotation character within a quoted string without an escape character."
```

Comment:
dafür gibts in Spock die `thrown()` Methode:

```
when:
writer.writeArray('"a";b')
then:
thrown(IllegalArgumentException)
```

Nur leider lässt sich da die Fehlermeldung net prüfen

Reply:
ja ich weiß, danke :). ich hab mich aber bewusst dafür entschieden die Fehlermeldung zu prüfen, da alle Exceptions den selben Typ haben und ich die sonst nicht unterscheiden kann

Reply:
hab nochmal geschaut, was gehen sollte wäre:
```
def e = thrown(IllegalArgumentException)
e.getMaessage() == ".." 
```